src/adapter/src/notice.rs

```diff
@@ -425,6 +431,20 @@ impl fmt::Display for AdapterNotice {
             AdapterNotice::PerReplicaLogRead { log_names } => {
                 write!(f, "Queried introspection relations: {}. Unlike other objects in Materialize, results from querying these objects depend on the current values of the `cluster` and `cluster_replica` session variables.", log_names.join(", "))
             }
+            AdapterNotice::VarDefaultUpdated { role, var_name } => {
+                let vars = match var_name {
+                    Some(name) => format!("variable {} was", name.quoted()),
+                    None => "variables were".to_string(),
+                };
+                let target = match role {
+                    Some(role_name) => role_name.quoted().to_string(),
+                    None => "the system".to_string(),
+                };
+                write!(
+                    f,
+                    "{vars} updated for {target}, please reconnect to take effect"
```

Comment:
I think we should be more descriptive and less prescriptive. Something along the lines of "This will have no effect on the current session" instead of "please reconnect to take effect".  The change may have been for some other role, or not relevant to the current session, etc.

Reply:
Good thought, updated!